components/sidebar/sidebar_category/sidebar_category.tsx

```diff
@@ -4,33 +4,112 @@
 import React from 'react';
 
 import SidebarChannel from '../sidebar_channel';
+import Constants, {A11yCustomEventTypes} from 'utils/constants';
+import {isKeyPressed} from 'utils/utils';
 
 type Props = {
     category: any;
     setChannelRef: (channelId: string, ref: HTMLDivElement) => void;
+    getChannelRef: (channelId: string) => HTMLDivElement | undefined;
+    actions: {
+        setCollapsedState: (categoryId: string, isCollapsed: boolean) => void;
+    };
 };
 
 type State = {
-
+    isCollapsed: boolean;
 };
 
 export default class SidebarCategory extends React.PureComponent<Props, State> {
+    categoryRef: React.RefObject<HTMLDivElement>;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.categoryRef = React.createRef();
+        this.state = {
+            isCollapsed: props.category.collapsed,
+        };
+    }
+
+    componentDidMount() {
+        // Refs can be null when this component is shallowly rendered for testing
+        if (this.categoryRef.current) {
+            this.categoryRef.current.addEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
```

Comment:
Can you use `?.` here to inline the null check?

Reply:
I get this eslint error: `Expected an assignment or function call and instead saw an expression.eslint(no-unused-expressions)`

Reply:
I remember we had issues before with ESLint and that operator on the mobile app. Maybe it's the same thing happening here